server/controllers/contest.js

```diff
@@ -0,0 +1,88 @@
+const Contest = require("../models/Contest");
+const asyncHandler = require("express-async-handler");
+const ObjectId = require("mongoose").Types.ObjectId;
+
+// Given parameters passed in, create a contest
+exports.createContest = asyncHandler(async (req, res, next) => {
+    const userID = req.user.id
+    const { title, description, prizeAmount, deadlineDate } = req.body;
+    const contest = new Contest({
+        title,
+        description,
+        prizeAmount,
+        deadlineDate,
+        userID
+    });
+    try {
+        const result = await contest.save();
+        if (!result) {
+            return res.status(400).json({ status: "contest not saved!!" });
+        }
+        res.status(201).json({
+            status: "contest saved!!",
+            contest,
+        });
+    } catch (error) {
+        return res.status(500).json({ error });
+    }
+})
+
+// Given an ID and new parameters, update the contest for any future edits
+exports.updateContestById = asyncHandler(async (req, res, next) => {
+    const contestID = req.params.id;
+    const { title, description, prizeAmount, deadlineDate } = req.body;
+    try {
+        const updatedContest = await Contest.findByIdAndUpdate(contestID, {
+            title,
+            description,
+            prizeAmount,
+            deadlineDate,
+            userID
+        });
+        if (!updatedContest) {
+            return res
+                .status(404)
+                .json({ status: "contest doesn't exist in records!!" });
+        }
+        // updating contest
+        res.status(201).json({
```

Comment:
Please use a general status code like 200.
The status code 201 is for resource creation :)

Ditto throughout!

Reply:
Sure, I did it on my latest commit.
Thanks